### PR TITLE
[kf5notifications] new port

### PR DESCRIPTION
**Pull request by wrobelda**
**Describe the pull request**
   Adds new port.
   Depends on:
   - [x] #19439 
   - [x] #19816 
   - [x] #20213 
   - [x] #20229 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All official, plus `arm64-osx`

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Comment by hellozee** [comment]
This only fails in the CI, locally works. For some reason it looks into the wrong directory I suppose or it just be the CI is configured that way. Cause it refers to the location where snoretoast is installed before running the post install checks.

**Comment by wrobelda** [comment]
> This only fails in the CI, locally works. For some reason it looks into the wrong directory I suppose or it just be the CI is configured that way. Cause it refers to the location where snoretoast is installed before running the post install checks.

Same exact experience.

**Comment by NancyLi1013** [comment]
The failures are like this:
```
Installing in D:/packages/kf5notifications_x86-windows/debug. Run D:/buildtrees/kf5notifications/x86-windows-dbg/prefix.sh to set the environment for KNotifications.
CMake Error in src/CMakeLists.txt:
  Imported target "SnoreToast::SnoreToastActions" includes non-existent path

    "D:/packages/libsnoretoast_x86-windows/include/snoretoast"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in src/CMakeLists.txt:
  Imported target "SnoreToast::SnoreToastActions" includes non-existent path

    "D:/packages/libsnoretoast_x86-windows/include/snoretoast"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.

```

**Comment by hellozee** [comment]
It is not reproducible locally tbh, no idea if we are doing anything wrong or there is something wrong with the CI.

**Comment by dg0yt** [comment]
> CMake Error in src/CMakeLists.txt:
>   Imported target "SnoreToast::SnoreToastActions" includes non-existent path
> 
>     "D:/packages/libsnoretoast_x86-windows/include/snoretoast"
> 
>   in its INTERFACE_INCLUDE_DIRECTORIES.

and

> It is not reproducible locally tbh, no idea if we are doing anything wrong or there is something wrong with the CI.

It is a `packages` path. You may have this path locally after building the dependency snoretoast. But it may be missing after removal & re-install from cache.
Please check the exported config from snoretoast for references to `packages`.

**Comment by hellozee** [comment]
Ahh make sense, would check that.

**Comment by wrobelda** [comment]
> It is a `packages` path. You may have this path locally after building the dependency libsnoretoast. But it may be missing after removal & re-install from cache.
> Please check the exported config from libsnoretoast for references to `packages`.

This is more nuanced than that. Locally build Config.cmake has:
```
  INTERFACE_INCLUDE_DIRECTORIES "C:/vcpkg/packages/libsnoretoast_x64-windows-static/include/snoretoast"
```

The path is correct, and the headers are included in the cached package:
```
PS C:\vcpkg> cat .\installed\vcpkg\info\libsnoretoast_0.8.0_x64-windows-static.list
x64-windows-static/
x64-windows-static/include/
x64-windows-static/include/snoretoast/
x64-windows-static/include/snoretoast/config.h
x64-windows-static/include/snoretoast/snoretoastactions.h
x64-windows-static/share/
x64-windows-static/share/libsnoretoast/
x64-windows-static/share/libsnoretoast/LibSnoreToastConfig-debug.cmake
x64-windows-static/share/libsnoretoast/LibSnoreToastConfig-release.cmake
x64-windows-static/share/libsnoretoast/LibSnoreToastConfig.cmake
x64-windows-static/share/libsnoretoast/copyright
x64-windows-static/share/libsnoretoast/vcpkg_abi_info.txt
x64-windows-static/tools/
x64-windows-static/tools/libsnoretoast/
x64-windows-static/tools/libsnoretoast/snoretoast.exe
```

File: ports/kf5notifications/fix_config_cmake.patch [hunk 1]
```diff
@@ -0,0 +1,64 @@
+From 1d3b1e0b70fe58f8bf2f1aac8f14536925ea0fa1 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Dawid=20Wro=CC=81bel?= <me@dawidwrobel.com>
+Date: Tue, 31 Aug 2021 20:19:56 +0200
+Subject: [PATCH] Add support for static builds
+
+---
+ KF5NotificationsConfig.cmake.in | 42 +++++++++++++++++++++++++++++++++
+ 1 file changed, 42 insertions(+)
+
```
Comment (NancyLi1013) [inline ports/kf5notifications/fix_config_cmake.patch:9]:
```suggestion
```
Could you please remove these message to minimize the patch?

File: ports/kf5notifications/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "kf5notifications",
+  "version-semver": "5.84.0",
```
Comment (NancyLi1013) [inline ports/kf5notifications/vcpkg.json:3]:
```suggestion
  "version": "5.84.0",
```
This might be `version `here.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/kf5notifications/fix_config_cmake.patch (hunk 1 above)
Comment (wrobelda) [reply ports/kf5notifications/fix_config_cmake.patch]: Done.

File: ports/kf5notifications/vcpkg.json (hunk 2 above)
Comment (wrobelda) [reply ports/kf5notifications/vcpkg.json]: Done.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for all the kf5Xxx ports!

**Pull request merged**